### ISSUE TITLE
Update method missing for Windows API - Customer

### ISSUE THREAD

**Issue by doughsay** [ruckus/quickeebooks#16]
I noticed the update methods are missing for the Windows side of the API.  I hope these will be implemented soon?

**Comment by ruckus** [comment]
Where I am right now with Quickeebooks is that I am implementing these methods as I need them, and the reality is is that I don't need update support for my apps right now.

But of course I am open to pull requests.

I know this isn't the answer you're after but it's the only one I got.

**Comment by ruckus** [comment]
Are you looking for a specific Object to update (e.g. Customer, Invoice, etc) - or just all? If its one or two I'd be willing to build them out, and then others could fill in some missing functionality. As you can see I dont even support all objects, I have 13 `Service` objects for Windows and only a subset of those even have `create` support.

**Comment by doughsay** [comment]
For the moment I'm just looking for Customer updates.  I'm not yet sure if I'll need anything else.  I see that someone else seems to have added this already in a fork: https://github.com/sillydeveloper/quickeebooks/commit/618ea1ab63b11833e4e1cebafe78e98782eebb4e

Haven't yet tried it.

Thanks for looking into this.

**Comment by ruckus** [comment]
@doughsay Ok I just added support for `Customer#update` in the Windows flavor. Can you please take a look?

Couple things to note:
1. Intuit does not support partial updates so any elements NOT provided will be **remove**.
2. You need to supply a `SyncToken` element with the updated value from the Intuit API. 

Thus, from my perspective its best if you first fetch the remote object by

`customer = customer_service.fetch_by_id(xx)`

and then apply your local changes to the result object, then provide that same object back to `update`.

If you build up a new `Customer` model object from scratch its likely you will not have the correct fields and Intuit is going to bitch.

Thus, if you read the object first you'll have the correct value for `SyncToken` sitting in memory and ready to be supplied back Intuit (amongst other attributes).

Does this make sense?

In summary, read first, apply changes and then update:

``` ruby

service = Quickeebooks::Windows::Service::Customer.new
service.access_token = oauth
service.realm_id = realm_id

customer = service.fetch_by_id(99)

# change the name
customer.name = "My Updated customer name"

# change other attributes here, like phones, email, etc.

service.update(customer)
```

**Comment by doughsay** [comment]
Great!  It works as advertised.  Thanks for looking into this.  I started myself, and discovered the same things you did, about intuit refusing the update with those fields set.  I managed to get it working, but had intermittent errors.  Yours seems to work without error so far.

I'll close this issue for now, if I end up adding any more update methods I'll send you a pull request.
